charts/endorser-service/templates/api/migration-job.yaml

```diff
@@ -0,0 +1,60 @@
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ include "endorser-service.fullname" . }}-migrate
+  labels:
+    {{- include "endorser-service.labels" . | nindent 4 }}
+  annotations:
+    "helm.sh/hook": post-install,post-upgrade
+    "helm.sh/hook-delete-policy": before-hook-creation,hook-succeeded
+spec:
+  backoffLimit: 2
+  template:
+    metadata:
+      labels:
+        {{- include "endorser-service.selectorLabels" . | nindent 8 }}
+    spec:
+      restartPolicy: Never
+      serviceAccountName: {{ include "endorser-service.serviceAccountName" . }}
+      initContainers:
+        - name: wait-for-db
+          image: busybox:1.36
+          command: ["/bin/sh","-c"]
+          args:
+            - >-
+              until nc -z {{ include "endorser-service.db.host" . }} {{ include "endorser-service.db.port" . }}; do echo waiting for db; sleep 2; done;
+
+      containers:
+        - name: migrate
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          command: ["/bin/sh","-c"]
+          args: ["python3 -m alembic upgrade head"]
```

Comment:
What's the benefit of having the migration happen as a separate job, rather than as part of the [entrypoint script](https://github.com/openwallet-foundation/acapy-endorser-service/blob/main/endorser/docker-entrypoint.sh) that comes with the Dockerfile? Mostly curious as I have not used a separate job for this purpose.

Reply:
The difference is: running the migration script every single time a pod is created or restarted. If there are multiple pods, then it runs multiple times, with every single pod on rollout or restart.
vs
Running it once per install/upgrade.

I think the latter option sounds safer.
If you agree, we should probably remove that from the entrypoint script, or make it skippable with an envvar when we build the proper image.

Otherwise, I can delete the job.